testapi.pm

```diff
@@ -527,20 +552,16 @@ sub click_lastmatch {
         last;
     }
 
+    my ($x, $y);
     # use center of the last area if no area contains click coordinates
-    if (!$relevant_area) {
-        $relevant_area = $last_matched_needle->{area}->[-1];
+    if (!$relevant_area and (!$relative_click_point || $relative_click_point eq "center")) {
+        ($x, $y) = _calculate_clickpoint($last_matched_needle);
     }
-    if (!$relative_click_point || $relative_click_point eq 'center') {
-        $relative_click_point = {
-            xpos => $relevant_area->{w} / 2,
-            ypos => $relevant_area->{h} / 2,
-        };
+    else {
```

Comment:
`$relevant_area` can still be `undef` or `"center"` here. Maybe these should be separate conditions?

Reply:
That's likely also the reason why the test fails.

Reply:
Ok, I will make the conditions separate, thanks.